src/Modules/Groups/components/GroupsData/GroupsData.tsx

```diff
@@ -0,0 +1,172 @@
+import axios, { AxiosError } from "axios";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { useLocation, useNavigate } from "react-router-dom";
+import useStudents from "../../../../Hooks/Students/useStudents";
+import { GROUPS_URLs, requestHeader } from "../../../../constans/END_POINTS";
+import { RequiredField } from "../../../../constans/VALIDATIONS";
+import { GroupFormData } from "../../../../interfaces/Groups/GroupFormData";
+import { toast } from "react-toastify";
+import { AxiosErrorResponse } from "../../../../interfaces/Authentication/AuthResponse";
+
+export default function GroupsData({
+  handleCloseAddUpdateModal,
+  getAllGroups,
+}: any) {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const status = location.state?.type === "edit";
+  const groupData = location.state?.GroupData;
```

Comment:
It is more common and reccomended to pass the id in a dynamic route or search params and get the data from the backend. This will enable us to share link with the same data. Currently, the state is just local and we will not be able to share the link with the same view.